index.js

```diff
@@ -11,19 +11,42 @@ var chroma = require('chroma-js');
 
 exports.main = function(){
 	var maxTabs = simplePrefs.prefs.maxTabs;
-	var max = (maxTabs > 1) ? maxTabs : 10; // Max at least 2 tabs please
+	var max = (maxTabs > 1) ? maxTabs : 2; // Max at least 2 tabs please
 	var maxPrivateBrowsing = simplePrefs.prefs.maxPrivateBrowsing;
+    var maxOpenInNextWindow = simplePrefs.prefs.maxOpenInNextWindow;
+    var maxExcludePinnedTabs = simplePrefs.prefs.maxExcludePinnedTabs;
 
 	var button = ActionButton({
 		id: 'max-tabs-button',
-	  label: title,
+	    label: title,
 		icon: './icon.svg',
 		disabled: true
 	});
 
+    var calcTabsLen = function(win){
+        let len = win.tabs.length;
+        if(maxExcludePinnedTabs){
+            let pinned = 0;
+            for(let i = 0; i < len; i++){
+                if(win.tabs[i].isPinned)
+                    pinned++;
+            }
+            len -= pinned;
```

Comment:
I get your point on performance :smiley: Compared to other Firefox addons, Max Tabs is fairly very small (for now, and I would like to keep it that way). Let's not over-optimize this while I prefer to make the code more readable and elegant for now :wink: